packages/block-library/src/table/style.scss

```diff
@@ -112,6 +112,58 @@
 		}
 	}
 
+	table[style*="border-top-color"] {
+		> *,
+		tr:first-child {
+			border-top-color: inherit;
+
+			th,
+			td {
+				border-top-color: inherit;
+			}
+		}
+
+		tr:not(:first-child) {
+			border-top-color: currentColor;
+		}
+	}
+	table[style*="border-right-color"] {
+		> *,
+		tr,
+		th,
+		td:last-child {
+			border-right-color: inherit;
+		}
+	}
+	table[style*="border-bottom-color"] {
+		> *,
+		tr:last-child {
+			border-bottom-color: inherit;
+
+			th,
+			td {
+				border-bottom-color: inherit;
+			}
+		}
+
+		// Border support classes and styles are applied on the table block
+		// itself. This means that without the rule below every table row would
+		// have a bottom border matching the color of the table's border.
+		// This style gives the best visual appearance and most expected result
+		// until we can control individual table rows or cells via inner blocks.
+		tr:not(:last-child) {
+			border-bottom-color: currentColor;
+		}
+	}
+	table[style*="border-left-color"] {
+		> *,
+		tr,
+		th,
+		td:first-child {
+			border-left-color: inherit;
+		}
+	}
+
```

Comment:
Further down below, do we also need to target individual border sides?

E.g., 

```css
	table[style*="border-top-width"],
	table[style*="border-bottom-width"],
	table[style*="border-right-width"],
	table[style*="border-left-width"],
	table[style*="border-width"] {
		> *,
		tr,
		th,
		td {
			border-width: inherit;
			border-style: inherit;
		}
	}
```

The frontend doesn't output shorthand so the borders aren't consistent. 

#### Editor
<img width="500" alt="Screen Shot 2022-04-13 at 10 12 00 am" src="https://user-images.githubusercontent.com/6458278/163075800-5c77399f-1ce3-45e5-87de-575d04bc968b.png">

#### Frontend
<img width="500" alt="Screen Shot 2022-04-13 at 10 11 44 am" src="https://user-images.githubusercontent.com/6458278/163075812-5bf26e1d-a799-4c1a-8cab-67f70a0d690a.png">

Reply:
Thanks @ramonjd for catching this. 🕵️ 

The editor forcing the shorthand styles for `border-width` meant that incorrect styling was applied to the inner borders within the editor. Until we have inner blocks for columns, rows, and cells, we don't have a neat means of styling any "inner borders" for the table. Instead, the idea here is if individual side borders are configured they are for the table as a whole i.e. the outermost top, right, bottom, and left borders.

I wasn't able to find a better solution at the moment other than adding a new class to the table within the _editor only_ and [using that to enforce the correct styling](https://github.com/WordPress/gutenberg/pull/37770/commits/ee1573a3d3ea3176d48996346f7a0290372c7fc1). It works as now the frontend and editors are consistent but I'm open to better solutions if you have any ideas.